zul/src/archive/web/js/zul/inp/NumberInputWidget.js

```diff
@@ -88,6 +88,14 @@ zul.inp.NumberInputWidget = zk.$extends(zul.inp.FormatWidget, {
 	},
 	getType: function () {
 		return zk.mobile && !this._format && !this._locale ? 'number' : this._type;
+	},
+	domAttrs_: function (no) {
+		var attr = this.$supers('domAttrs_', arguments);
+		if ((!no || !no.text) && zk.ipad)
```

Comment:
Why only iPad? Do you have any clue that Apple makes these differences?

Reply:
The issue I discuss with hawk 2 weeks ago, if add the 'pattern="[0-9]*"' at iphone or ipod, it will appear the number keyboard but without decimals point "." and basic operator "+ /". So the feature only works on ipad.